ts/src/profiler.ts

```diff
@@ -201,29 +254,37 @@ export class Profiler extends common.ServiceObject {
     const delayMillis = await this.collectProfile();
 
     // Schedule the next profile.
-    setTimeout(this.runLoop.bind(this), delayMillis).unref();
+    if (typeof delayMillis === 'number') {
```

Comment:
This dynamic cast on the type of the value returned from our own code is somewhat weird and should not be needed. Think about how to organize the code to avoid this.

Reply:
Reorganized to avoid this. Done.